crates/sol-types/src/types/calls.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{Panic, Result, Revert, SolError};
+use alloc::vec::Vec;
+
+/// A collection of ABI-encoded call-like types. This currently includes
+/// [`SolCall`] and [`SolError`].
+///
+/// [`SolCall`]: crate::SolCall
+/// [`SolError`]: crate::SolError
+///
+/// ### Implementer's Guide
+///
+/// We do not recommend implementing this trait directly. Instead, we recommend
+/// using the [`sol`][crate::sol] proc macro to parse a Solidity contract
+/// definition.
+pub trait SolCalls: Sized {
```

Comment:
This is something like `SolInterface` no? It's an enum over all expected inputs/outputs

Reply:
yeah but it's implemented for both calls and custom errors, doesn't have a name in abigen either, other than the struct named `<contract_name>{Calls,Errors}`